libraries/ms-common/src/watchdog.c

```diff
@@ -0,0 +1,19 @@
+#include "watchdog.h"
+
+void prv_expiry_callback(SoftTimerId timer_id, void *context) {
+  WatchdogStorage *storage = (WatchdogStorage *)context;
+  storage->callback(storage->callback_context);
+}
+
+void watchdog_start(WatchdogStorage *storage, WatchdogTimeout timeout_ms,
```

Comment:
I think this should be `watchdog_init` based on other modules. Also it should return a statuscode and shouldn't be allowed to be initialized multiple times on the same storage, probably?